src/osdatahub/FeaturesAPI/features_api.py

```diff
@@ -46,12 +47,16 @@ class FeaturesAPI:
     }
 
     def __init__(self, key: str, product_name: str, extent: Extent,
-                 spatial_filter_type: str = "intersects"):
+                 spatial_filter_type: str = "intersects", filter_join_type: str = "And"):
+        if filter_join_type not in {"And", "Or"}:
+            raise ValueError("Invalid filter join type '%s' (must be 'And' or 'Or')" % filter_join_type)
+
         self.key = key
         self.product = product_name
         self.extent = extent
         self.filters = []
         self.__spatial_filter = SpatialFilterTypes.get(spatial_filter_type)
```

Comment:
We might want to have a more complex filtering system to allow for some mix of "and" and "or". 
For example, if we have 3 filters (A, B, C), we want to allow A && (B || C), or any other variants. 
Could even use the new Union | operator for python 3.10?

Reply:
I don't think I makes sense to implement the union operator in this case. However, I agree that the logical and/or operators would be useful to allow a more complicated combination of filters.